gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+            raise ValueError("at least one of corpus/id2word must be specified, to establish "
+                             "input space dimensionality. Corpus should be provided using the "
+                             "keyword argument corpus.")
+
+        if type(topic_model_kind) == str:
+            self.topic_model_kind = {
+                "lda": ldamodel.LdaModel,
+                "ldamulticore": ldamulticore.LdaMulticore
+            }[topic_model_kind]
+        else:
+            self.topic_model_kind = topic_model_kind
+
+        # Store some of the parameters
+        self.num_models = num_models
+        self.gensim_kw_args = gensim_kw_args
+
+        # some settings affecting performance
+        self.memory_friendly_ttda = memory_friendly_ttda
+        self.distance_workers = distance_workers
+
+        # this will provide the gensim api to the ensemble basically
+        self.classic_model_representation = None
+
+        # the ensembles state
+        self.random_state = utils.get_random_state(random_state)
+        self.sstats_sum = 0
+        self.eta = None
+        self.tms = []
+        # initialize empty topic term distribution array
+        self.ttda = np.empty((0, len(gensim_kw_args["id2word"])))
+        self.asymmetric_distance_matrix_outdated = True
+
+        # in case the model will not train due to some
+        # parameters, stop here and don't train.
+        if num_models <= 0:
+            return
+        if ("corpus" not in gensim_kw_args):
+            return
+        if gensim_kw_args["corpus"] is None:
+            return
+        if "iterations" in gensim_kw_args and gensim_kw_args["iterations"] <= 0:
+            return
+        if "passes" in gensim_kw_args and gensim_kw_args["passes"] <= 0:
+            return
+
+        logger.info("Generating {} topic models...".format(num_models))
+        # training
+        if ensemble_workers > 1:
+            # multiprocessed
+            self.generate_topic_models_multiproc(num_models, ensemble_workers)
+        else:
+            # singlecore
+            self.generate_topic_models(num_models)
+
+        self.generate_asymmetric_distance_matrix(workers=self.distance_workers,
+                                                 threshold=masking_threshold,
+                                                 method=masking_method)
+        self.generate_topic_clusters(epsilon, min_samples)
+        self.generate_stable_topics(min_cores)
+
+        # create model that can provide the usual gensim api to the stable topics from the ensemble
+        self.generate_gensim_representation()
+
+    def convert_to_memory_friendly(self):
+        """removes the stored gensim models and only keeps their ttdas"""
+        self.tms = []
+        self.memory_friendly_ttda = True
+
+    def generate_gensim_representation(self):
+        """creates a gensim-model from the stable topics
+
+        The prior for the gensim model, eta, Can be anything,
```

Comment:
```suggestion
        The prior for the gensim model, eta, can be anything,
```

Reply:
Docstring here refactored to be more clear, including making this change.